dbt/runner.py

```diff
@@ -33,312 +36,340 @@
-            profile, model.compiled_contents, model.name)
-        rows = cursor.fetchall()
+    adapter.create_table(
+        profile,
+        schema=node_cfg.get('target_schema'),
+        table=node_cfg.get('target_table'),
+        columns=dest_columns,
+        sort=node_cfg.get('updated_at'),
+        dist=node_cfg.get('unique_key'))
 
-        cursor.close()
+    # TODO move this to inject_runtime_config, generate archive SQL
+    # in wrap step. can't do this right now because we actually need
+    # to inspect status of the schema at runtime and archive requires
+    # a lot of information about the schema to generate queries.
+    template_ctx = context.copy()
+    template_ctx.update(node_cfg)
 
-        if len(rows) > 1:
-            raise RuntimeError(
-                "Bad test {name}: Returned {num_rows} rows instead of 1"
-                .format(name=model.name, num_rows=len(rows)))
+    env = jinja2.Environment()
+    select = env.from_string(
+        dbt.templates.SCDArchiveTemplate,
+        template_ctx
+    ).render(node_cfg)
 
-        row = rows[0]
-        if len(row) > 1:
-            raise RuntimeError(
-                "Bad test {name}: Returned {num_cols} cols instead of 1"
-                .format(name=model.name, num_cols=len(row)))
+    insert_stmt = dbt.templates.ArchiveInsertTemplate().wrap(
+        schema=node_cfg.get('target_schema'),
+        table=node_cfg.get('target_table'),
+        query=select,
+        unique_key=node_cfg.get('unique_key'))
 
-        return row[0]
+    env = jinja2.Environment()
+    node['wrapped_sql'] = env.from_string(
+        insert_stmt,
+        template_ctx
+    ).render(node_cfg)
 
+    result = adapter.execute_model(
+        profile=profile,
+        model=node)
 
-class ArchiveRunner(BaseRunner):
-    run_type = dbt.model.NodeType.Archive
+    return result
 
-    def pre_run_msg(self, model):
-        print_vars = {
-            "schema": self.adapter.get_default_schema(self.profile),
-            "model_name": model.name,
-        }
+def run_hooks(profile, hooks, context, source):
+    if type(hooks) not in (list, tuple):
+        hooks = [hooks]
 
-        output = ("START archive table {schema}.{model_name} "
-                  .format(**print_vars))
-        return output
+    print('hooks')
+    print(hooks)
 
-    def post_run_msg(self, result):
-        model = result.model
-        print_vars = {
-            "schema": self.adapter.get_default_schema(self.profile),
-            "model_name": model.name,
-            "info": "ERROR archiving" if result.errored else "OK created"
-        }
+    ctx = {
+        "target": profile,
+        "state": "start",
+        "invocation_id": context['invocation_id'],
+        "run_started_at": context['run_started_at']
+    }
 
-        output = "{info} table {schema}.{model_name} ".format(**print_vars)
-        return output
+    compiled_hooks = [
+        dbt.compilation.compile_string(hook, ctx) for hook in hooks
```

Comment:
hooks should also have access to their model context, eg. `this` or `var`. Does that happen here?